src/math.ts

```diff
@@ -0,0 +1,27 @@
+import type { Length } from './primitives'
+
+type GetTuple<
+  L extends number,
+  result extends any[] = [],
+> = result['length'] extends L ? result : GetTuple<L, [...result, any]>
+
+namespace Math {
+  export type Subtract<
+    A extends number,
+    B extends number,
+  > = GetTuple<A> extends [...infer U, ...GetTuple<B>] ? U['length'] : 0
+
+  export type IsPositive<T extends number> = `${T}` extends `-${number}`
```

Comment:
Technically 0 is neither positive/negative -- This would treat it as positive, right?
(I imagine it doesn't matter for our use case, but technically this should really be something like `IsNonNegative`

Reply:
We can change that later - already merged - as we are not exposing it in the public API